code/datums/components/window_smoothing.dm

```diff
@@ -52,6 +52,7 @@ GLOBAL_LIST_EMPTY(window_appearances)
 	reset_icon()
 	RegisterSignal(parent, COMSIG_ATOM_SET_SMOOTHED_ICON_STATE, PROC_REF(on_junction_change))
 	RegisterSignal(parent, COMSIG_ATOM_UPDATE_ICON, PROC_REF(update_icon))
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, PROC_REF(remove_smoothing))
```

Comment:
A: shouldn't remove it should attempt a refresh, failing to readd if the window is in nullspace
B: should be done as a bespoke proc, to make the pipeline clear. remove_smoothing isn't a SIGNAL_HANDLER

Reply:
oh right im dumb. I've fixed it